File: src/fuzzy/rule/consequent/classLabel/class_label_basic.py
from src.fuzzy.rule.consequent.classLabel.abstract_class_label import AbstractClassLabel


class ClassLabelBasic(AbstractClassLabel):
    def __init__(self, class_label):
        super().__init__(class_label)

    def __eq__(self, other):
        if not isinstance(other, ClassLabelBasic):
            return False
        return other.get_class_label_value() == self.get_class_label_value()

    def is_rejected(self):
        return self._class_label == self.get_rejected_class_label_value()

    def set_rejected(self):
        self.set_class_label_value(self.get_rejected_class_label_value())

    def copy(self):
        return ClassLabelBasic(self.get_class_label_value())

    def __str__(self):
        if self.get_class_label_value() is None:
            raise Exception("class label value is None")
        return f"{self.get_class_label_value():2d}"

File: src/mofgbmlpy/core/data/class_label/class_label_basic.cpp
#include "class_label_basic.hpp"

ClassLabelBasic::ClassLabelBasic(int class_label) : AbstractClassLabel(), class_label(class_label) {
}

ClassLabelBasic::ClassLabelBasic(const ClassLabelBasic& other) 
    : AbstractClassLabel(other), class_label(other.class_label) {
}

int ClassLabelBasic::get_class_label_value() const {
    return class_label;
}

void ClassLabelBasic::set_class_label_value(int new_val) {
    class_label = new_val;
}

ClassLabelBasic::operator std::string() const
{
    return std::to_string(class_label);
}

bool ClassLabelBasic::operator==(const AbstractClassLabel& other) const
{
    const ClassLabelBasic* other_basic = dynamic_cast<const ClassLabelBasic*>(&other);
    if (!other_basic) {
        return false;
    }
    return class_label == other_basic->class_label;
}

ClassLabelBasic* ClassLabelBasic::clone() const
{
    return new ClassLabelBasic(*this);
}

